src/sandstorm/cgroup2.c++

```diff
@@ -0,0 +1,39 @@
+#include <sandstorm/cgroup2.h>
+#include <sandstorm/util.h>
+
+#include <kj/debug.h>
+
+namespace sandstorm {
+  Cgroup::Cgroup(kj::StringPtr path)
+    : dirfd(raiiOpen(path, O_DIRECTORY|O_CLOEXEC))
+  {}
+
+  Cgroup::Cgroup(kj::AutoCloseFd&& dirfd)
+    : dirfd(kj::mv(dirfd))
+  {}
+
+  Cgroup Cgroup::getOrMakeChild(kj::StringPtr path) {
+    int status;
+    do {
+      errno = 0;
+      status = mkdirat(dirfd.get(), path.cStr(), 0700);
+    } while(status != 0 && errno == EINTR);
```

Comment:
Maybe use `KJ_SYSCALL_HANDLE_ERRORS`, which takes care of EINTR handling and will let you handle EEXIST...

Reply:
Aha, knew something like that had to exist. Will do.